xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -4440,7 +4440,11 @@ bool CVideoPlayer::OnAction(const CAction &action)
     }
   }
 
-  if (dynamic_cast<CDVDInputStreamPVRManager*>(m_pInputStream))
+  //If the current channel is a nullptr, and this is a PVRStream, then this is a saved recording
+  CDVDInputStreamPVRManager* pvrStream = dynamic_cast<CDVDInputStreamPVRManager*>(m_pInputStream);
+  CPVRChannelPtr channel(g_PVRManager.GetCurrentChannel());
```

Comment:
Remove this line. See my other comment. <code>channel</code> value is no longer needed unconditionally. Calculate it only, if actually needed (performance!).